File: dcp-solutions/2024-12-25/solution.php
```php
<?php
function can_reach_end($arr) {
    $maxReach = 0;
    foreach ($arr as $i => $hops) {
        if ($i > $maxReach) return false;
        $maxReach = max($maxReach, $i + $hops);
    }
    return true;
}
?>
```

File: dcp-solutions/2024-12-25/solution.rb
```ruby
def can_reach_end(hops)
  max_reach = hops[0]
  for i in 1..hops.size - 1 do
    unless hops[i] + i > max_reach
      max_reach = [max_reach, i + hops[i]].max
    end
  end
  max_reach >= hops.size - 1
end
```